In [3]:
import numpy as np
import pandas as pd

house_data = pd.read_csv('./train.csv', header = None)
house_data.head

<bound method NDFrame.head of         0           1         2            3        4       5      6   \
0       Id  MSSubClass  MSZoning  LotFrontage  LotArea  Street  Alley   
1        1          60        RL           65     8450    Pave    NaN   
2        2          20        RL           80     9600    Pave    NaN   
3        3          60        RL           68    11250    Pave    NaN   
4        4          70        RL           60     9550    Pave    NaN   
...    ...         ...       ...          ...      ...     ...    ...   
1456  1456          60        RL           62     7917    Pave    NaN   
1457  1457          20        RL           85    13175    Pave    NaN   
1458  1458          70        RL           66     9042    Pave    NaN   
1459  1459          20        RL           68     9717    Pave    NaN   
1460  1460          20        RL           75     9937    Pave    NaN   

            7            8          9   ...        71      72     73  \
0     LotShape  LandC

In [5]:
print("Data Type : ", type(house_data)) #finding out what datatype the specified file is.
print("Data Dims : ", house_data.shape) #finding out the shape/dimensions of the datatype specified.

Data Type :  <class 'pandas.core.frame.DataFrame'>
Data Dims :  (1461, 81)


1b) The above returned results show that the csv contains 1461 rows, 1460 excluding the header while containing 81 columns, 80 excluding the index row.

In [6]:
house_data.info

<bound method DataFrame.info of         0           1         2            3        4       5      6   \
0       Id  MSSubClass  MSZoning  LotFrontage  LotArea  Street  Alley   
1        1          60        RL           65     8450    Pave    NaN   
2        2          20        RL           80     9600    Pave    NaN   
3        3          60        RL           68    11250    Pave    NaN   
4        4          70        RL           60     9550    Pave    NaN   
...    ...         ...       ...          ...      ...     ...    ...   
1456  1456          60        RL           62     7917    Pave    NaN   
1457  1457          20        RL           85    13175    Pave    NaN   
1458  1458          70        RL           66     9042    Pave    NaN   
1459  1459          20        RL           68     9717    Pave    NaN   
1460  1460          20        RL           75     9937    Pave    NaN   

            7            8          9   ...        71      72     73  \
0     LotShape  Lan

In [7]:
house_data.describe

<bound method NDFrame.describe of         0           1         2            3        4       5      6   \
0       Id  MSSubClass  MSZoning  LotFrontage  LotArea  Street  Alley   
1        1          60        RL           65     8450    Pave    NaN   
2        2          20        RL           80     9600    Pave    NaN   
3        3          60        RL           68    11250    Pave    NaN   
4        4          70        RL           60     9550    Pave    NaN   
...    ...         ...       ...          ...      ...     ...    ...   
1456  1456          60        RL           62     7917    Pave    NaN   
1457  1457          20        RL           85    13175    Pave    NaN   
1458  1458          70        RL           66     9042    Pave    NaN   
1459  1459          20        RL           68     9717    Pave    NaN   
1460  1460          20        RL           75     9937    Pave    NaN   

            7            8          9   ...        71      72     73  \
0     LotShape  L

In [9]:
summeroly2016data = pd.read_html('https://en.wikipedia.org/wiki/2016_Summer_Olympics_medal_table')
summeroly2016data

[                         2016 Summer Olympics medals  \
 0                                           Location   
 1                                         Highlights   
 2                                   Most gold medals   
 3                                  Most total medals   
 4  .mw-parser-output .hlist dl,.mw-parser-output ...   
 
                        2016 Summer Olympics medals.1  
 0                             Rio de Janeiro, Brazil  
 1                                         Highlights  
 2                                 United States (46)  
 3                                United States (121)  
 4  .mw-parser-output .hlist dl,.mw-parser-output ...  ,
                                                    0
 0                                Part of a series on
 1                               2016 Summer Olympics
 2  Bid process (bid details) Development (venues,...
 3  .mw-parser-output .navbar{display:inline;font-...,
                    Rank                   NOC  

In [10]:
len(summeroly2016data)

7

Therefore there are 7 tables in the imported html file. As len() returns the number of tables in the html file.

In [12]:
print("Data Type : ", type(summeroly2016data))

Data Type :  <class 'list'>


In [2]:
medaltbl = pd.DataFrame(summeroly2016data[2])
print(medaltbl)

NameError: name 'pd' is not defined

The pd.DataFrame function helps to make a table of the given data, you can specify which tables you want out of many
by using the indexing [] and the index of the table.

In [19]:
top20 = medaltbl[0:20]
print(top20)

   Rank            NOC  Gold  Silver  Bronze  Total
0     1  United States    46      37      38    121
1     2  Great Britain    27      23      17     67
2     3          China    26      18      26     70
3     4         Russia    19      17      20     56
4     5        Germany    17      10      15     42
5     6          Japan    12       8      21     41
6     7         France    10      18      14     42
7     8    South Korea     9       3       9     21
8     9          Italy     8      12       8     28
9    10      Australia     8      11      10     29
10   11    Netherlands     8       7       4     19
11   12        Hungary     8       3       4     15
12   13        Brazil*     7       6       6     19
13   14          Spain     7       4       6     17
14   15          Kenya     6       6       1     13
15   16        Jamaica     6       3       2     11
16   17        Croatia     5       3       2     10
17   18           Cuba     5       2       4     11
18   19    N

In [20]:
print("Data type : ", type(top20)) #top20 is a DataFrame type.

Data type :  <class 'pandas.core.frame.DataFrame'>
